pkg/component/networking/istio/istiod.go

```diff
@@ -265,6 +265,9 @@ func (i *istiod) Deploy(ctx context.Context) error {
 
 	registry := managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
 	for _, istioIngressGateway := range i.values.IngressGateway {
+		if "" != "TODO(istvanballok): remove this block once the issue: 'Istio metrics leak for deleted shoots' #12699 is resolved" {
```

Comment:
I guess you do not want to execute the code below, and also do not want to delete it for some reason, a better choice would be to use.

```suggestion
        // TODO(istvanballok): remove this block once the issue: 'Istio metrics leak for deleted shoots' #12699 is resolved
		if true {
```

Reply:
or 
define a variable 
```suggestion
skipIstioMetrics := true // TODO(istvanballok): remove this block once the issue: 'Istio metrics leak for deleted shoots' #12699 is resolved
```

Reply:
Thanks, I applied your suggestion.